app/components/team_sidebar/index.ts

```diff
@@ -1,45 +1,35 @@
 // Copyright (c) 2015-present Mattermost, Inc. All Rights Reserved.
 // See LICENSE.txt for license information.
 
-import {Q} from '@nozbe/watermelondb';
 import {withDatabase} from '@nozbe/watermelondb/DatabaseProvider';
 import withObservables from '@nozbe/with-observables';
 import {of as of$} from 'rxjs';
 import {switchMap} from 'rxjs/operators';
 
 import {Permissions} from '@constants';
-import {MM_TABLES, SYSTEM_IDENTIFIERS} from '@constants/database';
+import {queryRolesByNames} from '@queries/servers/role';
+import {queryMyTeams, queryOtherTeams} from '@queries/servers/team';
+import {observeCurrentUser} from '@queries/servers/user';
 import {hasPermission} from '@utils/role';
 
 import TeamSidebar from './team_sidebar';
 
 import type {WithDatabaseArgs} from '@typings/database/database';
-import type MyTeam from '@typings/database/models/servers/my_team';
-import type RoleModel from '@typings/database/models/servers/role';
-import type SystemModel from '@typings/database/models/servers/system';
-import type TeamModel from '@typings/database/models/servers/team';
-import type UserModel from '@typings/database/models/servers/user';
-
-const {SERVER: {SYSTEM, MY_TEAM, TEAM, USER, ROLE}} = MM_TABLES;
 
 const enhanced = withObservables([], ({database}: WithDatabaseArgs) => {
-    const currentUser = database.get<SystemModel>(SYSTEM).findAndObserve(SYSTEM_IDENTIFIERS.CURRENT_USER_ID).pipe(
-        switchMap(({value}) => database.get<UserModel>(USER).findAndObserve(value)),
-    );
-    const rolesArray = currentUser.pipe(
-        switchMap((u) => of$(u.roles.split(' '))),
+    const canCreateTeams = observeCurrentUser(database).pipe(
+        switchMap((u) => (u ? of$(u.roles.split(' ')) : of$([]))),
+    ).pipe(
+        switchMap((values) => queryRolesByNames(database, values).observe()),
+    ).pipe(
+        switchMap((r) => of$(hasPermission(r, Permissions.CREATE_TEAM, false))),
```

Comment:
no ned to change, but I think this can be written as

```suggestion
        switchMap((u) => (u ? of$(u.roles.split(' ')) : of$([]))),
        switchMap((values) => queryRolesByNames(database, values).observe()),
        switchMap((r) => of$(hasPermission(r, Permissions.CREATE_TEAM, false))),
```